docs/src/memory.md

```diff
@@ -0,0 +1,99 @@
+# Memory Management
+
+Unlike the [design of fiona](http://toblerity.org/fiona/manual.html#introduction), `ArchGDAL` does not automatically copy data from her data sources. This introduces concerns about memory management (whether objects should be managed by Julia's garbage collector, or by manually destroying the corresponding GDAL object).
+
+Currently this package provides data types corresponding to GDAL's Data Model, e.g.
+```julia
+mutable struct ColorTable;                    ptr::GDALColorTable         end
+mutable struct CoordTransform;                ptr::GDALCoordTransform     end
+mutable struct Dataset;                       ptr::GDALDataset            end
+mutable struct Driver;                        ptr::GDALDriver             end
+mutable struct Feature;                       ptr::GDALFeature            end
+mutable struct FeatureDefn;                   ptr::GDALFeatureDefn        end
+mutable struct FeatureLayer;                  ptr::GDALFeatureLayer       end
+mutable struct Field;                         ptr::GDALField              end
+mutable struct FieldDefn;                     ptr::GDALFieldDefn          end
+mutable struct Geometry <: AbstractGeometry;  ptr::GDALGeometry           end
+mutable struct GeomFieldDefn;                 ptr::GDALGeomFieldDefn      end
+mutable struct RasterAttrTable;               ptr::GDALRasterAttrTable    end
+mutable struct RasterBand;                    ptr::GDALRasterBand         end
+mutable struct SpatialRef;                    ptr::GDALSpatialRef         end
+mutable struct StyleManager;                  ptr::GDALStyleManager       end
+mutable struct StyleTable;                    ptr::GDALStyleTable         end
+mutable struct StyleTool;                     ptr::GDALStyleTool          end
+```
+and makes it the responsibility of the user to free the allocation of memory from GDAL, by calling `ArchGDAL.destroy(obj)` (which sets `obj.ptr` to `C_NULL` after destroying the GDAL object corresponding to `obj`).
+
+## Manual versus Context Management
+
+There are two approaches for doing so.
+
+1. The first uses the [`unsafe_` prefix](http://docs.julialang.org/en/release-0.4/manual/style-guide/#don-t-expose-unsafe-operations-at-the-interface-level) to indicate methods that returns objects that needs to be manually destroyed.
```

Comment:
https://docs.julialang.org/en/v0.6.2/manual/style-guide/#Don't-expose-unsafe-operations-at-the-interface-level-1 is a more up to date version.